timed/src/lib.rs

```diff
@@ -91,60 +127,99 @@ impl Trace {
         }
     }
 
-    pub fn dump(id: &str, processor: fn(&str), stats: &TracingStats) {
-        let start = std::time::Instant::now();
+    pub fn dump(id: &str, options: &TraceOptions) {
```

Comment:
I guess you can make it take &self and stop sending these params. Just a recommendation, not compulsory. Not sure if there is a performance impact.